pkg/proc/gdbserial/gdbserver.go

```diff
@@ -331,7 +331,9 @@ func (p *gdbProcess) Connect(conn net.Conn, path string, pid int, debugInfoDirs
 				// Workaround for darwin arm64. Apple's debugserver seems to have a problem
 				// with not selecting the correct thread in the 'g' command and the returned
 				// registers are empty / an E74 error is thrown.
-				if v[len("version:"):] == "1200" && p.bi.Arch.Name == "arm64" {
+				version, err := strconv.ParseInt(v[len("version:"):], 10, 32)
+
+				if err == nil && version >= 1200 && p.bi.Arch.Name == "arm64" {
```

Comment:
I don't think we should blacklist every version, I think apple should fix debugserver instead or change it to report the `g` command as unsupported. I support adding a check for `version == "1205"` as well as referencing the upstream bug: https://bugs.llvm.org/show_bug.cgi?id=50169

If they release another version without fixing the problem we'll blacklist it definitively.

Reply:
Suggestion: checking for `version >= 1200 && version <= 1205`?

Reply:
Ok. But I wonder why no one complained about version 1201. Does it even exist?

Reply:
I think it basically refers to the Xcode version, and yes 12.1, 12.2 and so do exist for Xcode at least.

Reply:
If that were the case, then why did no one complain?